### PR TITLE
MAINT: reviewer comment re: GPUs, etc.

### LINKED ISSUE #220: Comment to Address: Future of SciPy / GPU/ distributed ecosystem
I think @mdhaber will go ahead and add the majority of the reviewer comments from *Nature Methods* in **anonymous** fashion for the community to help address, as separate issues. Here's a first set of comments to address, grouped together because they are similar requests that we've been specifically asked to address by the Editors.

Reviewer 1:

> I would like a longer discussion about the future of SciPy particularly SciPy’s lack of the two features that are starting to become understood as essential to scientific computing: built-in automatic reverse-mode differentiation and heterogenous computation (e.g. CPU, DSP, GPU, etc.). Will SciPy adapt to support these increasingly vital features? Is it even possible without a substantial rewrite?

Reviewer 2:

> Distributed and GPU is considered out of scope but is conversely is believed to be required in the future for data and scientific analysis. A discussion of alternatives and how you interact with other ecosystems should be considered (e.g., RAPIDS, Dask, etc). How does SciPy progress with the rapidly evolving hardware?

### PR DESCRIPTION
**Pull request by tylerjereddy**
Fixes #220 

* both *Nature Methods* reviewers requested clarification
regarding SciPy's stance on staying relevant with GPUs
and distributed computing

* briefly mention that while these were out of scope
at the 1.0 release point, we now have an explicit
example of experimental GPU backend support in `scipy.fft`,
to be detailed in a future report

~We actually stopped testing PyPy on master, so that's a slightly awkward edit.~ This is no doubt contentious, but anyway let's make an effort and try to get it published soon :)

### CODE REVIEW

File: scipy-1.0/paper.tex [hunk 1]
```diff
@@ -618,12 +620,13 @@ \subsection*{Language choices}
 languages or major dependencies is generally unlikely---our choices are strongly
 driven by long-term stability. GPU acceleration, new transpiling libraries, and
 the latest JIT compilation approaches (i.e.,
-Numba\cite{Lam:2015:NLP:2833157.2833162}) are very powerful, but currently fall
+Numba\cite{Lam:2015:NLP:2833157.2833162}) are very powerful, but have traditionally fallen
 outside the remit of the main SciPy library. That said, we have recently
 increased our efforts to support compatibility with some of these options, and
-having our full test suite pass with the PyPy JIT
-compiler\cite{Bolz:2009:TMP:1565824.1565827} is now a requirement in our
-development workflow.
```
Comment (rgommers) [inline scipy-1.0/paper.tex]: Hmm yeah, I'd be inclined to leave this as it was true at the time of writing and for the 1.0 release. Or maybe tweak "is now a requirement". But our full test suite did pass with PyPy .....

File: scipy-1.0/paper.tex [hunk 2]
```diff
@@ -618,12 +620,13 @@ \subsection*{Language choices}
...
+while the exact approach we will adopt across the entire library remains unclear,
+we do now have a concrete implementation of a submodule that allows for the experimental
+use of multiple backends such as GPU-tractable data structures in the new
+\texttt{scipy.fft}. This will be described in detail in a future report.
```
Comment (rgommers) [inline scipy-1.0/paper.tex]: Is this the right place (Language choices section)? I'd expect such a sentence in the very last section of the paper.

File: scipy-1.0/paper.tex [hunk 3]
```diff
@@ -618,12 +620,13 @@ \subsection*{Language choices}
...
 the latest JIT compilation approaches (i.e.,
-Numba\cite{Lam:2015:NLP:2833157.2833162}) are very powerful, but currently fall
+Numba\cite{Lam:2015:NLP:2833157.2833162}) are very powerful, but have traditionally fallen
 outside the remit of the main SciPy library. That said, we have recently
 increased our efforts to support compatibility with some of these options, and
-having our full test suite pass with the PyPy JIT
-compiler\cite{Bolz:2009:TMP:1565824.1565827} is now a requirement in our
-development workflow.
+while the exact approach we will adopt across the entire library remains unclear,
+we do now have a concrete implementation of a submodule that allows for the experimental
```
Comment (VarIr) [inline scipy-1.0/paper.tex]: Does "submodule" refer to `scipy.fft`? I think "subpackage" would then be better for matters of consistency.

File: scipy-1.0/paper.tex (hunk 1 above)
Comment (tylerjereddy) [reply scipy-1.0/paper.tex]: Alright, I've restored the PyPy compatibility statement with an added 1.0 release point qualifier.

File: scipy-1.0/paper.tex (hunk 2 above)
Comment (tylerjereddy) [reply scipy-1.0/paper.tex]: That's fair, I've relocated to the final section of the manuscript.

File: scipy-1.0/paper.tex (hunk 3 above)
Comment (tylerjereddy) [reply scipy-1.0/paper.tex]: Alright, I've adopted "subpackage" instead of "submodule" now.

**Comment by tylerjereddy** [comment]
I think I've addressed the revisions requested so far. I'm happy to address any other concerns / adopt other suggestions.

File: scipy-1.0/paper.tex [hunk 4]
```diff
@@ -570,7 +570,9 @@ \subsection*{Project scope}
 In terms of software systems and architecture, SciPy's scope matches NumPy's:
 algorithms for in-memory computing on single machines, with support for a wide
 range of data types and process architectures. Distributed computing and support
-for graphics processing units (GPUs) are explicitly out of scope.
+for graphics processing units (GPUs) were explicitly out of scope at the 1.0
+release point, but this has been revised in our roadmap at the time of writing
+(see below).
```
Comment (mdhaber) [inline scipy-1.0/paper.tex]: Since it's far below, perhaps reference the discussion section.

File: scipy-1.0/paper.tex [hunk 5]
```diff
@@ -618,12 +620,11 @@ \subsection*{Language choices}
...
 the latest JIT compilation approaches (i.e.,
-Numba\cite{Lam:2015:NLP:2833157.2833162}) are very powerful, but currently fall
+Numba\cite{Lam:2015:NLP:2833157.2833162}) are very powerful, but have traditionally fallen
 outside the remit of the main SciPy library. That said, we have recently
 increased our efforts to support compatibility with some of these options, and
-having our full test suite pass with the PyPy JIT
-compiler\cite{Bolz:2009:TMP:1565824.1565827} is now a requirement in our
-development workflow.
+our full test suite passed with the PyPy JIT compiler\cite{Bolz:2009:TMP:1565824.1565827}
+at the 1.0 release point.
```
Comment (mdhaber) [inline scipy-1.0/paper.tex]: 👍

File: scipy-1.0/paper.tex (hunk 4 above)
Comment (tylerjereddy) [reply scipy-1.0/paper.tex]: changed to "see Discussion," thanks

**Comment by rgommers** [comment]
LGTM as is, merging. Thanks @tylerjereddy, all!

**Pull request merged**